src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy

```diff
@@ -106,38 +120,69 @@ class Liberty implements Plugin<Project> {
+        project.task('libertyDebug', type: DebugTask) {
             description 'Runs the Liberty Profile server in the console foreground after a debugger connects to the debug port (default: 7777).'
             logging.level = LogLevel.INFO
+            
+            project.afterEvaluate {
+                dependsOn installAppsDependsOn(server, 'installLiberty')
+            }
         }
 
         project.task('deploy', type: DeployTask) {
             description 'Deploys a supported file to the WebSphere Liberty Profile server.'
             logging.level = LogLevel.INFO
+            dependsOn 'libertyStart'
         }
 
         project.task('undeploy', type: UndeployTask) {
-            description 'Removes an application from the WebSphere Liberty Profile server.'
-            logging.level = LogLevel.INFO
+             description 'Removes an application from the WebSphere Liberty Profile server.'
+             logging.level = LogLevel.INFO
+             dependsOn 'libertyStart'
         }
 
-        project.task('installFeature', type: InstallFeatureTask, dependsOn: 'installLiberty') {
+        project.task('installFeature', type: InstallFeatureTask) {
             description 'Install a new feature to the WebSphere Liberty Profile server'
             logging.level = LogLevel.INFO
+
+            project.afterEvaluate {
+                if (project.liberty.features.name == null) {
+                    dependsOn 'libertyCreate'
+                } else {
+                    dependsOn 'installLiberty'
+                }
+            }
         }
-        project.task('uninstallFeature', type: UninstallFeatureTask, dependsOn: 'installLiberty') {
+        project.task('uninstallFeature', type: UninstallFeatureTask) {
             description 'Uninstall a feature from the WebSphere Liberty Profile server'
             logging.level = LogLevel.INFO
+            dependsOn 'libertyCreate'
         }
 
         project.task('cleanDirs', type: CleanTask) {
             description 'Deletes files from some directories from the WebSphere Liberty Profile server'
-            logging.level = LogLevel.INFO
+            logging.level = LogLevel.DEBUG
+
+            project.afterEvaluate {
+                if (project.liberty.cleanDir.workarea || project.liberty.cleanDir.dropins) {
+                    logger.info ('Stopping the server to clean workarea or dropins')
+                    dependsOn 'libertyStop'
+                }
+            }
         }
 
-        project.task('installApps', type: InstallAppsTask, dependsOn: 'libertyCreate') {
+        project.task('installApps', type: InstallAppsTask) {
             description "Copy applications generated by the Gradle project to a Liberty server's dropins or apps directory."
             logging.level = LogLevel.INFO
+            dependsOn project.tasks.withType(War)
+ 
+            project.afterEvaluate {
+                if (project.liberty.features.name != null) {
```

Comment:
Features moved to the server closure too.  Talk to @mattbsox if you are not sure what I mean here.

Reply:
If the features closure is not set for a server, will `features be null?

Reply:
done